packages/product/src/services/product-tag.ts

```diff
@@ -27,30 +32,96 @@ export default class ProductTagService<
     this.productTagRepository_ = productTagRepository
   }
 
+  async retrieve(
+    productTagId: string,
+    config: FindConfig<ProductTypes.ProductTagDTO> = {},
+    sharedContext?: Context
+  ): Promise<TEntity> {
+    return (await retrieveEntity<
+      ProductTag,
+      ProductTypes.ProductTagDTO
+    >({
+      id: productTagId,
+      entityName: ProductTag.name,
+      repository: this.productTagRepository_,
+      config,
+      sharedContext,
+    })) as TEntity
+  }
+
+  @InjectManager("productTagRepository_")
```

Comment:
question: On some `list`-methods we are adding this decorator, and on others, we are not. I am not exactly sure when this is required?

Reply:
AFAICT, it's required in the module/gateway service, but not in internal services. So is this just a minor blunder?

Reply:
This is required pretty much in every method that is "contextable", i.e. both gateway service and internal service methods. Every list, listAndCount and retrieve methods should be wrapped in this decorator. 

This will come as a follow up PR since it'll pollute this PR even more and requires changing up tests of all the other services.

Reply:
Got it thanks!

Reply:
I've added them, can fix the specs later. (https://github.com/medusajs/medusa/pull/4535/commits/1d019083185d053323902dee7983ba783f901d99)